### PR TITLE
cAdvisor 0.9.0

### PR DESCRIPTION
**Pull request by vmarmol**

### CODE REVIEW

**Comment by vmarmol** [comment]
Ping @vishh @rjnagal all integration tests pass on our usual setups.

**Comment by rjnagal** [comment]
Can we check resource usage on a systemd setup? I was wondering if reporting cpu load hierarchically makes dynamic housekeeping slowdown less likely.

**Comment by vmarmol** [comment]
From my tests 3-6% on a system with O(30) containers. Mainly 3-4% with a
spike every min to 6%. I think that's within bounds, we may need to do an
optimization round in the near future.

**Comment by rjnagal** [comment]
Do we know what it was with 0.8.0?

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,12 @@
 # Changelog
 
+## 0.9.0 (2015-04-06)
+- Added CPU load information.
+- Support for more network devices.
+- Support for more partition types.
+- More robust handling of stats failures.
```
Comment (rjnagal) [inline CHANGELOG.md]: I think we also started reporting diskio stats.

Comment (vmarmol) [inline CHANGELOG.md]: Ah good point, added.

**Comment by vmarmol** [comment]
Will test and report back. We should report usage as part of the release candidates.

**Comment by vmarmol** [comment]
Uuuu, ouch. So the current usage is ~1% with minute spikes to 3%. That's definitely a big jump in CPU usage. I will try to verify if this is due to CPU load. If so, we should add a flag to allow it to be disabled for the users that worry about that.

You made a good point about the load maybe making it less likely that housekeeping doesn't get lowered. I will check into that as well.

**Comment by rjnagal** [comment]
If the load is causing the housekeeping to change, we should just disable
it rather than add a new flag.
I think we can fix that easily by changing the dynamic housekeeping check
to look at cpu and memory usage specifically rather than comparing the
whole set.

**Comment by vmarmol** [comment]
Regathered the numbers and removed the UI from the equation:
- 0.8.0: ~0.7%
- 0.9.0 without CPU load: ~1.2%
- 0.9.0 with load: ~1.5% (there are bigger peaks more often than in the two above though)

Dynamic housekeeping did not seem to affect the results. So we're using roughly twice as much CPU. Digging into why without CPU load the usage went up so high.

**Comment by vmarmol** [comment]
Alright, found the culprits. So relative to 0.8.0:
- +0.4% CPU usage for Disk IO stats
- +0.3% CPU usage for CPU load stats

**Comment by vmarmol** [comment]
Removed load, PTAL.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,13 @@
 # Changelog
 
+## 0.9.0 (2015-02-06)
+- Support for more network devices.
+- Support for more partition types.
```
Comment (vishh) [inline CHANGELOG.md]: Can we be specific here instead of more? I guess we support btrfs in addition to existing "ext4" support?

**Comment by rjnagal** [comment]
LGTM

**Comment by vishh** [comment]
LGTM excepting for that one comment.

**Comment by vmarmol** [comment]
Done, merge after the other PRs.

**Comment by rjnagal** [comment]
merging...

**Pull request merged**